service/history/outbound_queue_factory.go

```diff
@@ -145,7 +159,10 @@ func NewOutboundQueueFactory(params outboundQueueFactoryParams) QueueFactory {
 					RunnableFactory: func(e queues.Executable) ctasks.Runnable {
 						key := grouper.KeyTyped(e.GetTask())
 						return ctasks.RateLimitedTaskRunnable{
-							Limiter: rateLimiterPool.Get(key),
+							Limiter: rateLimiterPool.Get(queues.NamespaceIDAndDestination{
+								NamespaceID: key.NamespaceID,
+								Destination: key.Destination,
+							}),
```

Comment:
Please revert this change, I'll address grouping of cancel and start requests properly later.